### DISCUSSION TITLE
Animated border-bottom

### DISCUSSION THREAD

**Discussion by iTheMaverick** [community/community#71819] [Programming Help]
### Body

Hi, so I have a border-bottom that appears when hovering over navbar items and was curious if there is a way to animate these using CSS, nothing crazy just maybe like a linear slide from left to right instead of it just making the border appear.

`<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <link rel="stylesheet" href="CSS/contactus.css" />
    />

    <title>Toplist Perfumes</title>
  </head>
  <body>
    <header>
      <h1>Contact Us</h1>
      <hr class="Topbar" />
      <navbar class="Topnav">
          <a class="Home" href="index.html">Home</a>
          <a class="NewArrivals" href="newarrivals.html">New Arrivals</a>
            <a class="BrandCollection" href="brandcollection.html">Brand Collection</a>
              <a class="Catalog" href="catalog.html">Catalog</a>
                <a class="ContactUs" href="contactus.html">Contact Us</a>
                  <a class="FinanceYourPurchase" href="financeyourpurchase.html">Finance Your Purchase</a>
<a class="Login" href="login.html">Login</a>
<a class="Search" href="search.html">Search</a>
<a class="Cart" href="cart.html">Cart</a>
        <img
        class="HeaderLogo"
        src="img/TopListScentsWebsite.png"
        alt="ToplistLogo"
        />
      </navbar>
    </header>
    <div>
      <h2 class="About">About Toplist Perfumes</h2>
      <p class="Info">
        835 N Michigan Ave, Chicago, IL, 60611 Unit 6005<br><br
        >Phone: +1 847-497-8459 <br>Email: Toplistscents@gmail.com</br><br>Monday-Friday 11am-7pm<br>Saturday-Sunday 11am–6pm
      </p>
      <hr class="Divider">
      <h3>Careers</h3>
      <p class="Careers">For careers, contact Toplistscents@gmail.com</p>
      <hr class="Divider">
      <h4 >Inquiries</h4>
      <p class="Inquiries">Contact Toplistscents@gmail.com</p>
    </div>
    <footer>
      <hr class="BottomBar" />
      <navbar class="BottomNav">
        <a class="SearchBottom" href="search.html">Search</a>
        <a class="LegalPolicies" href="LegalPolicies.html">Legal Policies</a>
          <a class="RefundPolicy" href="RefundPolicy.html">Refund Policy</a>
            <a class="PrivacyPolicy" href="PrivacyPolicy.html">Privacy Policy</a>
              <a class="AboutUs" href="aboutus.html">About Us</a>
              <hr class="BottomBarTwo" />
              <p class="Copyright">© 2023 Toplist, All rights reserved</p>
    </navbar>
    </footer>
  </body>
</html>
`

`* {
  padding: 0;
  margin: 0;
  box-sizing: 0;
}

body {
  font-family: sans-serif;
  background-color: #ffffff;
}

.Topnav {
  margin: auto;
  padding-top: 2%;
  position: fixed;
  text-align: center;
  font-size: 1.8vh;
  white-space: nowrap;
}

.HeaderLogo {
  position: fixed;
  padding: auto;
  margin: auto;
  width: 20%;
  left: 3%;
  top: 0.5%;
}

.Topbar {
  position: fixed;
  width: 100%;
  height: 13%;
  right: 0%;
  background: #ffffff;
  top: 0%;
  padding: auto;
  margin: auto;
  border: none;
  border-bottom: 1px solid #1a1a1a;
}

.Home {
  margin: auto;
  color: #1a1a1a;
  text-decoration: none;
  margin-right: 2%;
  margin-left: 70%;
}

.Home:hover {
  border-bottom: 1px solid #1a1a1a;
  transition: border-bottom 0.6s linear;
}

.NewArrivals {
  margin: auto;
  color: #1a1a1a;
  text-decoration: none;
  margin-right: 2%;
}

.NewArrivals:hover {
  border-bottom: 1px solid #1a1a1a;
}

.BrandCollection {
  margin: auto;
  color: #1a1a1a;
  text-decoration: none;
  margin-right: 2%;
}

.BrandCollection:hover {
  border-bottom: 1px solid #1a1a1a;
}

.Catalog {
  margin: auto;
  color: #1a1a1a;
  text-decoration: none;
  margin-right: 2%;
}

.Catalog:hover {
  border-bottom: 1px solid #1a1a1a;
}

.ContactUs {
  color: #1a1a1a;
  text-decoration: none;
  border-bottom: 1px solid #1a1a1a;
  margin-right: 2%;
}

.ContactUs:hover {
  border-bottom: 1px solid #1a1a1a;
}

.FinanceYourPurchase {
  color: #1a1a1a;
  text-decoration: none;
  margin-right: 2%;
}

.FinanceYourPurchase:hover {
  border-bottom: 1px solid #1a1a1a;
}

.Login {
  color: #1a1a1a;
  text-decoration: none;
  margin-left: 40%;
}

.Login:hover {
  border-bottom: 1px solid #1a1a1a;
}

.Search {
  color: #1a1a1a;
  text-decoration: none;
  margin-left: 2%;
}

.Search:hover {
  border-bottom: 1px solid #1a1a1a;
}

.Cart {
  color: #1a1a1a;
  text-decoration: none;
  margin-left: 2%;
}

.Cart:hover {
  border-bottom: 1px solid #1a1a1a;
}

h1 {
  position: absolute;
  padding-top: 8%;
  padding-left: 45%;
  margin: auto;
  font-size: 4vh;
  text-align: center;
  font-family: system-ui;
  font-weight: 350;
}

h2 {
  position: relative;
  margin-top: 11%;
  margin-left: 25%;
  font-size: 3vh;
  text-align: left;
  font-family: system-ui;
  font-weight: 350;
}

.Info {
  position: relative;
  padding-top: 1%;
  padding-left: 25%;
  margin: auto;
  font-size: 2vh;
  font-family: system-ui;
  font-weight: 100;
}

.Careers {
  position: relative;
  padding-top: 1%;
  padding-left: 25%;
  margin: auto;
  font-size: 2vh;
  font-family: system-ui;
  font-weight: 100;
}

.Inquiries {
  position: relative;
  padding-top: 1%;
  padding-left: 25%;
  padding-bottom: 2%;
  margin: auto;
  font-size: 2vh;
  font-family: system-ui;
  font-weight: 100;
}

h3 {
  position: relative;
  padding-top: 1%;
  padding-left: 25%;
  margin: auto;
  font-size: 3vh;
  text-align: left;
  font-family: system-ui;
  font-weight: 350;
}

h4 {
  position: relative;
  padding-top: 1%;
  padding-left: 25%;
  margin: auto;
  font-size: 3vh;
  text-align: left;
  font-family: system-ui;
  font-weight: 350;
}

.Divider {
  margin-left: 15%;
  margin-right: 15%;
  margin-top: 2%;
  opacity: 30%;
}

.BottomBar {
  position: fixed;
  width: 100%;
  height: 20%;
  right: 0%;
  background: #1a1a1a;
  bottom: 0%;
  padding: auto;
  margin: auto;
  border: none;
  border-top: 5px solid #1a1a1a;
}

.BottomNav {
  margin: auto;
  padding-top: 3.5%;
  position: fixed;
  text-align: center;
  font-size: 1.8vh;
  white-space: nowrap;
  font-family: system-ui;
}

.SearchBottom {
  color: #ffffff;
  text-decoration: none;
  margin-right: 2%;
  margin-left: 70%;
}

.LegalPolicies {
  color: #ffffff;
  text-decoration: none;
  margin-right: 2%;
}

.RefundPolicy {
  color: #ffffff;
  text-decoration: none;
  margin-right: 2%;
}

.PrivacyPolicy {
  color: #ffffff;
  text-decoration: none;
  margin-right: 2%;
}

.AboutUs {
  color: #ffffff;
  text-decoration: none;
  margin-right: 2%;
}

.BottomBarTwo {
  position: fixed;
  text-align: center;
  width: 75%;
  right: 15%;
  bottom: 5%;
  opacity: 10%;
  padding: auto;
  margin: auto;
  border: none;
  border-top: 1px solid #ffff;
}

.Copyright {
  position: fixed;
  color: #ffffff;
  text-decoration: none;
  font-size: 1.3vh;
  font-family: system-ui;
  bottom: 2%;
  left: 10%;
}
`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by deep100x** [comment]
Add this code to your existing CSS

--> Apply a transition effect to the border-bottom property

.Topnav a {
border-bottom: 1px solid transparent; 
transition: border-bottom 0.3s ease; 
}
.Topnav a:hover {
border-bottom: 1px solid #1a1a1a; / Set the border color to the desired color */
}

❌
* {         
padding: 0;
margin: 0;
box-sizing: 0;
}
✔box-sizing : border-box;

Instead of using too much Margin & Padding Use Flex Properties.

Instead of giving everywhere { text-decoration : none; }  do this  put it in  Universal * { . }
